cmake/app/boilerplate.cmake

```diff
@@ -274,6 +274,36 @@ set(KERNEL_EXE_NAME   ${KERNEL_NAME}.exe)
 set(KERNEL_STAT_NAME  ${KERNEL_NAME}.stat)
 set(KERNEL_STRIP_NAME ${KERNEL_NAME}.strip)
 
+# Try to populate USER_HOME_DIR with a directory that user
```

Comment:
On Linux, the latest freedesktop.org standards prefer the following, in order:

XDG_CACHE_HOME
$HOME/.cache
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

On macOS, ~/Library/Caches is preferred:

https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1

Reply:
Done